Add median operation to Math blocks

AdityaByte opened the pull request:
<!--
Thanks for contributing a pull request to MIT App Inventor. Please answer the following questions to help us review your changes.
-->

## General items:

- [ ] I have updated the relevant documentation files under docs/
- [ ] My code follows the:
    - [ ] [Google Java style guide](https://google.github.io/styleguide/javaguide.html) (for .java files)
    - [x] [Google JavaScript style guide](https://google.github.io/styleguide/jsguide.html) (for .js files)
- [ ] `ant tests` passes on my machine

<!--
This section pertains to changes to the components module that affect the code running on the Android device.
-->

## Device/runtime changes:

- [ ] I branched from `ucr`
- [ ] My pull request has `ucr` as the base

<!--
Further, if you've changed the blocks language or another user-facing designer/blocks API
-->

## Blocks language / API changes:

- [ ] I have updated the corresponding version number in
      `appinventor/components/src/.../common/YaVersion.java`
- [ ] I have updated the corresponding upgrader in
      `appinventor/appengine/src/.../client/youngandroid/YoungAndroidFormUpgrader.java`
- [ ] I have updated the corresponding entries in
      `appinventor/blocklyeditor/src/versioning.js`

<!--
This section pertains to changes that affect appengine, blocklyeditor (except changes to block semantics),
buildserver, components (but not changes to runtime), and docs.
-->

## Other changes:

- [x] I branched from `master`
- [x] My pull request has `master` as the base

---

## What does this PR accomplish?

This PR adds a **`median`** operation to the Math blocks in MIT App Inventor.

The new block allows users to compute the median of a list of numeric values, correctly handling both odd- and even-length lists. While it is currently possible to calculate the median using existing blocks, the implementation is unwieldy; this PR provides a built-in, reusable solution.

### Implementation details:
- Added a new block definition to the Blockly Math category
- Implemented YAIL code generation for the median operation
- Added runtime Scheme support for both Android and iOS

### Notes:
When testing locally, I encountered the following runtime error:
`Argument (package median, Unknown, version 0.0)
to 'apply-to-args' has wrong type (java.lang.Package)`

This may be related to symbol binding or naming in the Scheme runtime. I would appreciate guidance on whether any required exports or conventions are missing.

---

Resolves #3458

AdityaByte committed:
Implemented median statistic in Math component

- Rename runtime function to yail-median
- Update block language version and versioning table
- Add documentation to markdown source

ewpatton commented:
@AppInventorWorkerBee ok to test

ewpatton commented on appinventor/components-ios/src/runtime.scm:
```diff
@@ -2338,6 +2209,26 @@ Matrix implementation.
 )
 
+;;; Calculate the median of the list
+(define (median l)
```
Ditto previous comment about `yail-median`.

ewpatton reviewed (changes requested):
A preliminary review. I haven't yet tested the functionality.

ewpatton commented on appinventor/components-ios/src/runtime.scm:
```diff
@@ -2338,6 +2209,26 @@ Matrix implementation.
+      (signal-runtime-error
+        "The list cannot be empty to find the median."
+        "List is empty")
+      (let* ((sorted-list (sort l-content <))
```
See my comment on the Android runtime.scm. It is equally valid here.

AdityaByte commented:
@ewpatton 
Thanks for the review!
I’ve updated the implementation to use `yail-list-sort` instead of `SRFI-32` sort, aligned the iOS version with Android, and fixed the help URL anchor to `#median` to match the Markdown documentation.

daki7711 commented:
@AdityaByte would you be willing to solve these merge conflicts? if you are too busy I can take on it from here
